watcher/watcher.go

```diff
@@ -1218,18 +1281,29 @@ func (w *Watcher) podsOfIncorrectVersion(
 	scheduler *models.Scheduler,
 ) ([]v1.Pod, error) {
 	incorrectPods := []v1.Pod{}
-	for _, pod := range pods.Items {
-		podMajorVersion, _, err := w.splitedVersion(pod.Labels["version"])
+
+	schedulerMajorVersion, _, err := w.splitedVersion(scheduler.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	schedulerMajorRollbackVersion := -1
+	if scheduler.RollbackVersion != "" {
+		schedulerMajorRollbackVersion, _, err = w.splitedVersion(scheduler.RollbackVersion)
 		if err != nil {
 			return nil, err
 		}
+	}
 
-		schedulerMajorVersion, _, err := w.splitedVersion(scheduler.Version)
+	for _, pod := range pods.Items {
+		podMajorVersion, _, err := w.splitedVersion(pod.Labels["version"])
 		if err != nil {
 			return nil, err
 		}
 
-		if podMajorVersion != schedulerMajorVersion {
+		// if a rollback happened consider 2 versions as correct:
```

Comment:
This can be confusing. Is this really necessary? Couldn't we use only the actual version(after DB rollback)?

Reply:
I agree it is confusing, but if I only consider the actual version it will replace all pods (even the ones that are in the correct version) if a rollback occur. If I consider the version that was rolled back to, it will only restart pods that need to

Reply:
I can see the tradeoff here. In terms of code simplicity I'd go with replacing all pods, but I can see that this could cause unnecessary load and instability for the game. So I am okay with the approach you proposed.